using a concept of "related to" results in app exception thrown

this is occurring in bl-lookup app versions 1.1.6, 1.1.7 and 1.1.8.

example: 
https://bl-lookup-sri.renci.org/bl/related%20to?version=latest or https://bl-lookup-sri.renci.org/bl/related%20to?version=2.2.5

either example results in a 503 error on the client. each request generates differing stack dumps:

[2022-01-27 16:08:14 +0000] [1] [ERROR] Exception occurred while handling uri: 'http://bl-lookup-sri.renci.org/bl/biolink:related_to?version=latest'
Traceback (most recent call last):
ERROR:sanic.root:Exception occurred while handling uri: 'http://bl-lookup-sri.renci.org/bl/biolink:related_to?version=latest'
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sanic/app.py", line 939, in handle_request
    response = await response
  File "/home/nru/bl_lookup/server.py", line 55, in properties
    return response.json(props)
  File "/usr/local/lib/python3.9/site-packages/sanic/response.py", line 210, in json
    dumps(body, **kwargs),
TypeError: Annotation(tag='biolink:canonical_predicate', value='True', extensions={}, annotations={}) is not JSON serializable
  File "/usr/local/lib/python3.9/site-packages/sanic/app.py", line 939, in handle_request
    response = await response
  File "/home/nru/bl_lookup/server.py", line 55, in properties
    return response.json(props)
  File "/usr/local/lib/python3.9/site-packages/sanic/response.py", line 210, in json
    dumps(body, **kwargs),
TypeError: Annotation(tag='biolink:canonical_predicate', value='True', extensions={}, annotations={}) is not JSON serializable
ERROR:sanic.root:Exception occurred while handling uri: 'http://bl-lookup-sri.renci.org/bl/related_to?version=2.2.5'
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sanic/app.py", line 939, in handle_request
    response = await response
  File "/home/nru/bl_lookup/server.py", line 55, in properties
    return response.json(props)
  File "/usr/local/lib/python3.9/site-packages/sanic/response.py", line 210, in json
  
  
[2022-01-27 16:08:51 +0000] [1] [ERROR] Exception occurred while handling uri: 'http://bl-lookup-sri.renci.org/bl/related_to?version=2.2.5'
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/sanic/app.py", line 939, in handle_request
    response = await response
  File "/home/nru/bl_lookup/server.py", line 55, in properties
    return response.json(props)
  File "/usr/local/lib/python3.9/site-packages/sanic/response.py", line 210, in json
    dumps(body, **kwargs),
TypeError: Annotation(tag='biolink:canonical_predicate', value='True', extensions={}, annotations={}) is not JSON serializable
    dumps(body, **kwargs),
TypeError: Annotation(tag='biolink:canonical_predicate', value='True', extensions={}, annotations={}) is not JSON serializable

--- a/bl_lookup/bl.py
+++ b/bl_lookup/bl.py
@@ -146,6 +146,14 @@ class bmt_wrapper():
             del element['local_names']
         if 'slot_usage' in element:
             del element['slot_usage']
+        #Annotations are also not JSON serializable.  The current structure is:
+        # 'annotations': annotations={'biolink:canonical_predicate':
+        # Annotation(tag='biolink:canonical_predicate', value='True', extensions={}, annotations={})}
+        # And we want to turn that into "Tag":"Value"
+        if 'annotations' in element:
+            for k,v in element['annotations'].items():
+                element[v['tag']] = v['value']
+            del element['annotations']
         return element
 
     def get_descendants(self, name):

--- a/bl_lookup/server.py
+++ b/bl_lookup/server.py
@@ -198,8 +198,8 @@ async def resolve(request):
                     #Can't invert something with no inverse.
                     inverted = False
                 else:
-                    annots = props['annotations']
-                    if (annots is not None) and ('biolink:canonical_predicate' in annots) and annots['biolink:canonical_predicate'].value:
+                    #annots = props['annotations']
+                    if 'biolink:canonical_predicate' in props and props['biolink:canonical_predicate'].upper() == 'TRUE':
                         #this is the canonical direction, all good
                         inverted = False
                     else:

--- a/tests/test_service.py
+++ b/tests/test_service.py
@@ -369,6 +369,17 @@ def test_lookup_with_commas():
         assert (response.status_code == 200)
 
 
+def test_lookup_related_to():
+    # setup some parameters
+    param = {'version': '2.2.5'}
+
+    # make a good request
+    #request, response = app.test_client.get('/bl/related_to', params=param)
+    request, response = app.test_client.get('/bl/acts_upstream_of', params=param)
+
+    # was the request successful
+    assert(response.status_code == 200)
+
 def test_lookup_lineage():
     # setup some parameters
     param = {'version': '2.2.3'}
@@ -458,7 +469,7 @@ def test_properties():
     ret = json.loads(response.body)
 
     # check the data
-    assert(len(ret) == 54 and ret['id_prefixes'][0] == 'PUBCHEM.COMPOUND' and ret['class_uri'] == 'biolink:SmallMolecule' and 'is metabolite' in ret['slots'])
+    assert(len(ret) == 53 and ret['id_prefixes'][0] == 'PUBCHEM.COMPOUND' and ret['class_uri'] == 'biolink:SmallMolecule' and 'is metabolite' in ret['slots'])
 
     # make a bad request
     request, response = app.test_client.get('/bl/bad_substance', params=param)

